web/src/lib/roadmap/roadmapBuilder.ts

```diff
@@ -109,7 +109,7 @@ const addTasksFromAddOn = (builder: RoadmapBuilder, addOns: AddOn[]): RoadmapBui
     if (!task) continue;
     builder.tasks = [
       ...builder.tasks,
-      { filename: task, weight: addOn.weight, stepNumber: addOn.step, required: addOn.required },
+      { filename: task, weight: addOn.weight, stepNumber: addOn.step, required: !!addOn.required },
```

Comment:
While the `RoadmapBuilder` type enforces the `required` field to be a `boolean`, when the roadmap is generated from the builder downstream, the `Roadmap` type allows it to be `undefined` again. Maybe worth looking into converting that over to just a `boolean` at some point.

Reply:
You could also just switch this to an explicit test, like so:

```typescript
{
  required: addOn.required === true
}
```

Reply:
True. `Boolean(addOn.required)` is another option but I figured `!!` is the more common approach to Boolean conversion. Any limitations to sticking with that?

Reply:
Nope. If eslint doesn't error, you're fine!